src/daft-plan/src/logical_ops/sample.rs

```diff
@@ -0,0 +1,28 @@
+use std::sync::Arc;
+
+use crate::LogicalPlan;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Sample {
+    // Upstream node.
+    pub input: Arc<LogicalPlan>,
+    pub fraction: String,
```

Comment:
because of the trait bounds `#[derive(Clone, Debug, PartialEq, Eq, Hash)]` i couldn't store fraction as a f64, so i stored it as a string instead, then later on in physical plan I parse it as a f64. Is there a better way to do this?

Reply:
You should be able to `impl Eq for Sample` and set the behavior for comparing the f64.

Reply:
got it, did the impl for Eq and Hash